service/history/transferQueueActiveTaskExecutor.go

```diff
@@ -235,7 +236,21 @@ func (t *transferQueueActiveTaskExecutor) processWorkflowTask(
 	// release the context lock since we no longer need mutable state builder and
 	// the rest of logic is making RPC call, which takes time.
 	release(nil)
-	return t.pushWorkflowTask(ctx, task, taskQueue, timestamp.DurationFromSeconds(taskScheduleToStartTimeoutSeconds))
+
+	err = t.pushWorkflowTask(ctx, task, taskQueue, timestamp.DurationFromSeconds(taskScheduleToStartTimeoutSeconds))
+
+	if _, ok := err.(*serviceerrors.StickyWorkerUnavailable); ok {
+		// sticky worker is unavailable, switch to original task queue
+		taskQueue = &taskqueuepb.TaskQueue{
+			// do not use task.TaskQueue which is sticky, use original task queue from mutable state
+			Name: mutableState.GetExecutionInfo().TaskQueue,
+			Kind: enumspb.TASK_QUEUE_KIND_NORMAL,
+		}
+		workflowRunTimeout := timestamp.DurationValue(executionInfo.WorkflowRunTimeout)
+		taskScheduleToStartTimeoutSeconds = int64(workflowRunTimeout.Round(time.Second).Seconds())
+		err = t.pushWorkflowTask(ctx, task, taskQueue, timestamp.DurationFromSeconds(taskScheduleToStartTimeoutSeconds))
```

Comment:
nit: it took me sometime to realize why calling reset stickyness api is unnecessary here. maybe add some comments explaining the reasoning?

Reply:
added comments